gnd/src/main/java/com/google/android/gnd/persistence/local/room/ElementEntity.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.local.room;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.Index;
+import androidx.room.PrimaryKey;
+import com.google.android.gnd.model.form.Element;
+import com.google.android.gnd.model.form.Element.Type;
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.AutoValue.CopyAnnotations;
+
+@AutoValue
+@Entity(
+    tableName = "element",
+    foreignKeys =
+        @ForeignKey(
+            entity = FormEntity.class,
+            parentColumns = "id",
+            childColumns = "form_id",
+            onDelete = ForeignKey.CASCADE),
+    indices = {@Index("form_id"), @Index("field_id")})
+public abstract class ElementEntity {
+
+  @PrimaryKey(autoGenerate = true)
+  public int id;
```

Comment:
We're using String uuids for elements, so you'd want to remove autoGenerate and change this to a string here.

Reply:
Elements didn't have a separate id of it's own.

Ideally, it would be cleaner if I add FieldEntity as `@Embedded` in this entity. But since we are keeping the options open regarding element types, I chose not to.

But having an autoGenerate primary key isn't useful and is causing duplicate entries. I need to fix this

Reply:
> Elements didn't have a separate id of their own.

They should.. where did you see that they don't? New Elements are never created on the Android side, their uuid is set by the form designer when they're added there. The schema is defined by example here:
https://github.com/google/ground-platform/wiki/Cloud-Firestore-Representation#document-structure

Reply:
I got confused because the Element model didn't have an `id` field. I'll add that to both places

Reply:
I think we can remove `ElementEntity` and store `FieldEntity` directly for each element. It does not contain any unique information anyways.